contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -283,22 +292,40 @@ def _transcribe(item, parents, editRate, masterMobs):
         # The goal here is to find an available range. Media ranges are stored
         # in the related MasterMob, and there should only be one - hence the name
         # "Master" mob. Somewhere down our chain (either a child or our parents)
-        # is a MasterMob. For SourceClips in the CompositionMob, it is our child.
-        # For everything else, it is a previously encountered parent. Find the
-        # MasterMob in our chain, and then extract the information from that.
-        child_mastermob = (
-            item.mob if isinstance(item.mob, aaf2.mobs.MasterMob) else None
-        )
+        # is a MasterMob.
+        # There are some special cases where the masterMob could be:
+        # 1) For SourceClips in the CompositionMob, the mob the SourceClip is
+        #    referencing can be our MasterMob.
+        # 2) If the source clip is referencing another CompositionMob,
+        #    drill down to see if the composition holds the MasterMob
+        # 3) For everything else, it is a previously encountered parent. Find the
+        #    MasterMob in our chain, and then extract the information from that.
+
+        # Additionally, regards metadata (UserComments), they are usually stored
+        child_mastermob, compositionUserMetadata = _find_mastermob_for_sourceclip(item)
+
+        if compositionUserMetadata:
+            metadata['UserComments'] = compositionUserMetadata
+
         parent_mastermobs = [
             parent for parent in parents
             if isinstance(parent, aaf2.mobs.MasterMob)
         ]
         parent_mastermob = parent_mastermobs[0] if len(parent_mastermobs) > 1 else None
+
+        if child_mastermob:
+            transcribe_log("[found child_mastermob]", indent)
+        elif parent_mastermob:
+            transcribe_log("[found parent_mastermob]", indent)
+        else:
+            transcribe_log("[found no mastermob]", indent)
+
         mastermob = child_mastermob or parent_mastermob or None
 
         if mastermob:
-            # get target path
-            mastermob_child = masterMobs.get(str(mastermob.mob_id))
+            # Get target path
+            mastermob_child = _transcribe(mastermob, list(), editRate, indent)
```

Comment:
Can a MasterMob be referenced multiple times? If so, is there an opportunity to cache the result of this _transcribe? If that seems tricky, we could just leave a TODO or file an issue for later.

Reply:
Usually there is one mastermob associated with a SourceClip as far as my experience goes. I don't think caching the master-mob will gain us much. But I can just test it with some of our large sequences and see if I get the same master-mob-id several times.

Reply:
The original intent of the mastermob list was meant to avoid re-transcribing the same mastermobs again and again. I was under the (somewhat naive) impression that this would be common. 

In your testing so far, is the adapter faster, slower, or about the same as before this change? If it is the same or faster with a few real world AAFs then, we can leave this until another time when we do some more extensive performance testing. If it is slower, then we should look deeper since folks are already unhappy with the performance.

Reply:
I would not have removed the first pass if it would have made the adapter slower. The reason (besides simplifying the code) is a 30-40% speed gain (also listed in the original commit-description), tested on three large sequences from three different shows of ours.

Reply:
Faster is great! Thanks for checking.